File: src/main/java/aurocosh/divinefavor/common/block/soulbound_lectern/SoulboundLecternState.java
package aurocosh.divinefavor.common.block.soulbound_lectern;

import net.minecraft.util.IStringSerializable;

public enum SoulboundLecternState implements IStringSerializable {
    UNBOUND("unbound"),
    BOUND("bound");

    // Optimization
    public static final SoulboundLecternState[] VALUES = SoulboundLecternState.values();

    private final String name;

    SoulboundLecternState(String name) {
        this.name = name;
    }

    @Override
    public String getName() {
        return name;
    }
}

File: src/main/java/aurocosh/divinefavor/common/block/soulbound_lectern/SoulboundLecternState.kt
package aurocosh.divinefavor.common.block.soulbound_lectern

import net.minecraft.util.IStringSerializable

enum class SoulboundLecternState private constructor(private val value: String) : IStringSerializable {
    INACTIVE("inactive"),
    ACTIVE("active");

    override fun getName(): String {
        return value
    }

    companion object {

        // Optimization
        val VALUES = SoulboundLecternState.values()
    }
}